perllib/FixMyStreet/DB/Result/User.pm

```diff
@@ -451,6 +451,25 @@ sub adopt {
     $other->delete;
 }
 
+sub anonymize_account {
+    my $self = shift;
+
+    $self->problems->update({ anonymous => 1, name => '', send_questionnaire => 0 });
+    $self->comments->update({ anonymous => 1, name => '' });
+    $self->alerts->update({ whendisabled => \'current_timestamp' });
+    $self->password('', 1);
+    $self->update({
+        email => 'removed-' . $self->id . '@' . FixMyStreet->config('EMAIL_DOMAIN'),
+        email_verified => 0,
+        name => '',
+        phone => '',
+        phone_verified => 0,
+        title => undef,
+        twitter_id => undef,
+        facebook_id => undef,
```

Comment:
I wonder if this should unset superuser and body things just to be on the safe side. Or in some other way special case those?

Reply:
Would you ever want to anonymize a staff/superuser account, even if inactive, good question. Note there should be no way of logging in with these accounts as email and phone verified are both 0, no password, and email won't be delivered (and even if it were, it would create a new account with the same email, verified). So it might be useful to know if an account was superuser/body without knowing who it was?